suite/tests/client-interface/ldstex.c

```diff
@@ -459,44 +459,28 @@ GLOBAL_LABEL(FUNCNAME:)
         add      w2, w2, #0x1
         stxp     w3, w1, w2, [x0]
         cbnz     w3, 2b
-        /* Test pair4-single8. */
```

Comment:
Even if some behavior is officially unpredictable, we would prefer to handle it if it's easy to do so, since we may well see it in the field. It is not unheard of to encounter apps that execute officially-unpredictable instructions, especially low-level hacky code doing obfuscation or malware.

What will happen with this PR if the pair-singleton-mismatch test cases here are run under DR?  The checks for matching sizes will fail and we'll have an infinite loop?  To add back handling without introducing mismatches again for matching app code: for split-block we'd need some complex handling?  What about same-block cases: is that easy?

OTOH, we have already decided to not support several cases of unpredictable behavior for ldstex: mismatched sizes and addresses.  Since we've already done that for ldstex maybe this is ok.  Could you edit api/docs/ldstex.dox as part of this PR to augment the section that begins "Our scheme above checks for strict equality" to also state that we require pair parity or matching pairness or however it is best explained (probably needs more words than that).

Reply:
> Even if some behavior is officially unpredictable, we would prefer to handle it if it's easy to do so

It turned out to be fairly easy to get it to handle the mismatched-number-of-registers case so I have rewritten the patch to fix the bug but keep support that case.

Reply:
OK, SGTM.